pkg/controller/service/service_controller.go

```diff
@@ -348,6 +359,9 @@ func (s *ServiceController) ensureLoadBalancer(service *v1.Service) (*v1.LoadBal
 	if err != nil {
 		return nil, err
 	}
+	if len(nodes) == 0 {
+		return nil, fmt.Errorf("There are no available nodes for LoadBalancer service %s/%s", service.Namespace, service.Name)
```

Comment:
Why should we return error here? I think this should be handled at cloud provider layer, in case cloud provider implementations might have different interpretations when there is no nodes (though yeah most of them might just return an similar error)?

Reply:
Yes, cloud provider can't ensure a valid load balancer when there are no available nodes.
Service controller just calls EnsureLoadBalancer() when nodes isn't empty.

Reply:
I still think we shouldn't return error at service controller layer. What if the cloud provider needs to cleanup resource or reconfigure LB when there is no nodes? It seems safer if cloud provider could be aware of this as well.

Reply:
> Service controller just calls EnsureLoadBalancer() when nodes isn't empty.

I'm not seeing this definition in the cloud interface spec: https://github.com/kubernetes/kubernetes/blob/1339e4cffcd343fde2458c4e1b81e364d0060bbb/pkg/cloudprovider/cloud.go#L94-L98

Reply:
Actually, OpenStack cloud provider need non-empty nodes at v1.7. I will check other cloud provider later.
All right, I will remove this code. Thank  you.